server/ai_live_video.go

```diff
@@ -86,16 +127,21 @@ func startTricklePublish(ctx context.Context, url *url.URL, params aiRequestPara
 			clog.V(8).Infof(ctx, "trickle publish writing data seq=%d", seq)
 			segment, err := publisher.Next()
 			if err != nil {
-				clog.Infof(ctx, "error getting next publish handle; dropping segment err=%v", err)
-				params.liveParams.sendErrorEvent(fmt.Errorf("Missing next handle %v", err))
+				stopProcessing(ctx, params, fmt.Errorf("error getting next publish handle; dropping segment, err=%w", err))
 				return
 			}
 			for {
+				select {
+				case <-ctx.Done():
+					stopProcessing(ctx, params, errors.New("stream processing is done"))
+					return
+				default:
+				}
+
 				startTime := time.Now()
 				currentSeq := slowOrchChecker.GetCount()
 				if seq != currentSeq {
-					clog.Infof(ctx, "Next segment has already started; skipping this one seq=%d currentSeq=%d", seq, currentSeq)
-					params.liveParams.sendErrorEvent(fmt.Errorf("Next segment has started"))
+					stopProcessing(ctx, params, fmt.Errorf("next segment has already started seq=%d currentSeq=%d", seq, currentSeq))
```

Comment:
Generally when we had a problem with the orchestrator with the old code, we'd disconnect the stream. In this PR, we change it to swap instead of disconnecting.

However, in the old code, this stream was not being disconnected here; it was looping back around to the next segment. Can we maintain that behavior here for now, instead of swapping? The same goes anywhere else we use `sendErrorEvent` instead of `stopPipeline` (eg, during that "error publishing" case below) 

Since this is already a big change on its own it would be good to preserve the old behavior where possible, without introducing *more* cases for swapping.

Reply:
Yeah, sure. Sorry, I didn't mean to change it, I probably just misunderstood this part. I'll change it back to "looping back around to the next segment".